src/main/java/carpetextra/helpers/ObsidianBlock.java

```diff
@@ -21,14 +21,11 @@ public ObsidianBlock(Settings settings)
     }
     
     @Override
-    public boolean hasRandomTicks(BlockState state)
+    public void randomTick(BlockState blockState_1, ServerWorld serverWorld_1, BlockPos blockPos_1, Random random_1)
     {
-        return CarpetExtraSettings.renewableLava;
-    }
+        if(!CarpetExtraSettings.renewableLava)
+            return;
```

Comment:
Why is this better? Haven't looked at random ticks for quite a while, but IIRC I think then random ticks to obsidian would count towards the limit, but not execute since they would just return (therefore resulting in non-vanilla behaviour). Would need checking though. Or was it failing else?

Reply:
Other vanilla blocks seems to do exactly what was here before, check for example `NetherWartBlock`

Reply:
I changed this because `hasRandomTicks()` is already set to true from `ticksRandomly()` in the block settings, but I suppose using it might be better

Reply:
`ticksRandomly` is a function that enables the randomticks for the `AbstractBlock.Settings` instance, which isn't called in obsidian's (default value being `false`.